zerver/views/messages.py

```diff
@@ -754,6 +750,7 @@ def get_messages_backend(request: HttpRequest, user_profile: UserProfile,
     if anchored_to_right:
         num_after = None
 
+    first_visible_message_id = get_first_visible_message_id(user_profile.realm)
```

Comment:
Reading the code for `get_first_visible_message_id`, I don't understand how the logic there is correct; it seems to assume that anything not present in the cache is not a restricted realm, but I don't believe that's a correct inference, since the general design of Zulip is that any cache can be cleared without harm.  What's the story here?

Reply:
@timabbott I think the idea was we that we can afford a cache miss occasionally as it just results in the removal of the message history restriction for some time. So it was thought off as a tradeoff for increasing the query speed.